### DISCUSSION TITLE
75. 함수 종속성의 개념과 종류를 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#77] [General]
각 함수 종속성은 어느 정규형 과정에서 처리되나요?

**Comment by eun-byeol** [comment]
함수 종속성은 `특정 속성이 다른 속성에 의해 유일하게 결정되는 관계`를 의미합니다. 함수 종속성은 `정규화 과정`에서 데이터 중복을 제거하는데 사용됩니다.
함수 종속성의 종류로는 `완전 함수 종속`, `부분 함수 종속`, `이행적 함수 종속`, `다치 종속`, `조합 종속`이 있습니다.

`완전 함수 종속`은 기본키를 구성하는 모든 속성에 종속된 경우입니다. `제2정규형`에서 부분 함수 종속을 제거하면서 만족됩니다.
`부분 함수 종속`은 기본키가 복합키일 때 속성 일부에 종속되는 경우입니다. `제2정규형`에서 제거됩니다.
`이행적 함수 종속`은 릴레이션에서 X가 Y를 결정하고, Y가 Z를 결정할 때 X가 Z를 결정하는 경우입니다. `제3정규형`에서 제거됩니다.
`다치 종속`은 X가 Y를 결정할 때, X의 값에 대해 Y의 값이 여러 개 존재할 수 있는 경우입니다. `제4정규형`에서 처리됩니다.
`다중 값 종속(조합 종속)`은 X와Y가 결합되어 Z를 결정하는 경우입니다. `제5정규형`에서 처리됩니다.

![image](https://github.com/user-attachments/assets/97cad333-2f0e-4a43-a072-d9310238e364)
